port/linux/omrvmem.c

```diff
@@ -961,52 +962,79 @@ default_pageSize_reserve_memory(struct OMRPortLibrary *portLibrary, void *addres
 	 * Any changes made here may need to be reflected in that version.
 	 */
 	int fd = -1;
+	int ft = -1;
 	int flags = MAP_PRIVATE;
 	void *result = NULL;
 	int protectionFlags = PROT_NONE;
+	BOOLEAN useBackingFile = FALSE;
 
 	Trc_PRT_vmem_default_reserve_entry(address, byteAmount);
 
+	if(mode & OMRPORT_VMEM_MEMORY_MODE_FILE_HANDLE) {
+		useBackingFile = TRUE;
+	} else {
 #if defined(MAP_ANONYMOUS)
-	flags |= MAP_ANONYMOUS;
+		flags |= MAP_ANONYMOUS;
+		useBackingFile = FALSE;
 #elif defined(MAP_ANON)
-	flags |= MAP_ANON;
+		flags |= MAP_ANON;
+		useBackingFile = FALSE;
 #else
-	fd = portLibrary->file_open(portLibrary, "/dev/zero", EsOpenRead | EsOpenWrite, 0);
-	if (-1 != fd)
+		useBackingFile = TRUE;
 #endif
-	{
-		if (0 != (OMRPORT_VMEM_MEMORY_MODE_COMMIT & mode)) {
-			protectionFlags = get_protectionBits(mode);
-		} else {
-			flags |= MAP_NORESERVE;
+	}
+
+	char filename[FILE_NAME_SIZE];
+	if (useBackingFile) {
+		char int_str[FILE_NAME_SIZE / 2 + 1];
+		sprintf(int_str, "%09d", getpid()); // Max pid in 64bit system is 4194304. Pads with zeros.
+		strcpy(filename, "tempfile");
+		strcat(filename, int_str);
+
+		fd = portLibrary->file_open(portLibrary, filename, EsOpenRead | EsOpenWrite | EsOpenCreate, 0);
+		identifier->fd = fd;
+		omrfile_unlink(portLibrary, filename);
```

Comment:
I dont think you should be calling the `macro` version here.  This should likely call through the portLibrary variable the same as file_open.